src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1123,6 +1125,10 @@ private function addSerializerSection(ArrayNodeDefinition $rootNode, callable $e
                         ->arrayNode('default_context')
                             ->normalizeKeys(false)
                             ->useAttributeAsKey('name')
+                            ->beforeNormalization()
+                                ->ifTrue(fn () => $this->debug && class_exists(JsonParser::class))
+                                ->then(fn (array $v) => $v + [JsonDecode::DETAILED_ERROR_MESSAGES => true])
+                            ->end()
                             ->defaultValue([])
```

Comment:
the default value should probably also include `JsonDecode::DETAILED_ERROR_MESSAGES => true` (when appropriate) so that this configuration also applies when no default_context is configured at all.

Reply:
see #57804